### DISCUSSION TITLE
Daylight saving time changeover

### DISCUSSION THREAD

**Discussion by linkworx-stijnvandenbroucke** [quartznet/quartznet#1639] [Q&A]
Hi,

Is there a way to reschedule the jobs when daylight saving time starts or stops?
I'm using a lot of cron, but it seems the first execution after dst had changed, the timestamp of execution is not correct. After the first execution, it runs normal.

I did a test by rescheduling the jobs based on the event 'SystemEvents.TimeChanged', but that event happens a few times per month.
I would like to limit to the moments it's really useful to do it.

Br, Stijn.

**Comment by lahma** [comment]
Before going into details how hard DST handling is, why do you need to schedule things around DST chnage time (usually those  morning hours). DST happens usually twice a year, is it such a big problem when the job is still actually executed, but might not be the exact time (that  someone observes).

**Reply by linkworx-stijnvandenbroucke** [reply]
Hi, the reason is that the applications has to run 24/24 7/7.
The job needs to trigger a record split in the database: period 1 stops and period 2 starts.
After DST change, the duration of the period is not ok. And that is very annoying in the reporting part.
That's why I'm looking for a solution...

**Reply by lahma** [reply]
Wouldn't it be easier to run tasks say at 6AM and correlate the correct stop/start period based on your needs? Isn't there a problem anyway with missing hours when DST causes some point of time being non-existent for given date ("03:15 doesn't exist at this date").

**Reply by linkworx-stijnvandenbroucke** [reply]
During the 'active period', IoT data is processed. If you don't store it correct when you receive it, it's not possible to 're-assign' to the correct period. 
There will be a missing hour, that is not the point. Data is collected per 8 hours: start 22h, stop 06h. The cron of 06h executes at 5h or 7h when DTS has changed.  And that causes a period from 22h til 05h h or 22h til 07h, so 7 or 9 hours in totals instead of 8.

**Reply by lahma** [reply]
So why not simple trigger with fixed 8h interval with your chosen start time. If 8h interval is the thing.

**Reply by linkworx-stijnvandenbroucke** [reply]
It's generic system based on a calendar, the 8 hours is an example.

**Reply by lahma** [reply]
Have you checked whether DailyTimeIntervalTrigger would work better in your case?

**Reply by lahma** [reply]
Also,  running cron trigger using UTC as timezone should give reliable results.
